src/lib/common/sol-platform-linux-common.c

```diff
@@ -776,3 +798,99 @@ sol_platform_linux_uevent_unsubscribe(const char *action, const char *subsystem,
 
     return 0;
 }
+
+
+int64_t
+sol_platform_impl_get_system_clock(void)
+{
+    return (int64_t)time(NULL);
+}
+
+static bool
+system_clock_changed(void *data, int fd, uint32_t active_flags)
+{
+    char buf[4096];
+
+    if (read(timer_ctx.fd, buf, sizeof(buf)) >= 0)
+        return true;
+    close(timer_ctx.fd);
+    timer_ctx.fd = -1;
+    timer_ctx.watcher = NULL;
+    sol_platform_register_system_clock_monitor();
+    sol_platform_inform_system_clock_changed();
+    return false;
+}
+
+int
+sol_platform_register_system_clock_monitor(void)
+{
+    int r;
+    struct itimerspec spec;
+
+    if (timer_ctx.watcher)
+        return 0;
+
+    timer_ctx.fd = timerfd_create(CLOCK_REALTIME, TFD_CLOEXEC);
+    SOL_INT_CHECK(timer_ctx.fd, < 0, -errno);
+
+    memset(&spec, 0, sizeof(struct itimerspec));
+    spec.it_value.tv_sec += 0xfffffff0;
```

Comment:
Why the magic value? Might be nice to have a comment explaining what the hack does.